src/Signer.php

```diff
@@ -16,14 +15,14 @@ public function getAlgorithmId(): string;
     /**
      * Creates a hash for the given payload
      *
-     * @throws InvalidArgumentException When given key is invalid.
+     * @throws InvalidArgument When given key is invalid.
      */
     public function sign(string $payload, Key $key): string;
 
     /**
      * Returns if the expected hash matches with the data and key
      *
-     * @throws InvalidArgumentException When given key is invalid.
+     * @throws InvalidArgument When given key is invalid.
```

Comment:
In this case, I believe we had many things under the same exception type:

1. Unable to read the file contents
1. Key isn't parseable by OpenSSL
1. Key isn't compatible (ECDSA vs RSA keys)
1. Invalid signature length for ECDSA

Reply:
Don't have proper suggestions atm, though (shaaame)